content/docs/command-reference/add.md

```diff
@@ -148,6 +148,8 @@ directory symlinks.
   pattern as specified by `target`. Shell-style wildcards are supported: `*`,
   `?`, `[seq]`, `[!seq]`, and `**`.
 
+- `--desc` - arbitrary string description for generated `.dvc` file.
```

Comment:
for generated .dvc file -> for data artifact ... or something similar, it does not describe `.dvc` file.

I wonder if we should provide a short-cut (`-d`)? also, would `-m` be a better option? to remind `git commit -m` ?

Reply:
it should be also `--desc <description>` I think?

Reply:
> for generated .dvc file -> for data artifact ... or something similar, it does not describe .dvc file.

Depends on how to look at it. It is not really for specific `out` either, from the internal perspective. In 2.0 we'll likely simplify the internal structure for `dvc add`-generated file, and then your suggestion will be 100% correct.

> I wonder if we should provide a short-cut (-d)? also, would -m be a better option? to remind git commit -m ?

Both -d and -m are taken. Don't want to take up short options for it just yet. Also analogy with `git commit -m` is not that intuitive for this, IMO.

> it should be also --desc <description> I think?

Added. Thanks.

Reply:
> Depends on how to look at it. 

it's important to understand what is the purpose of this from the end user perspective and describe it in a way that would resonates with that in the first place. To my mind it's about providing descriptions to data unless I'm missing something. Now I read as I describe the file - why would I even want to do that?

> Don't want to take up short options for it just yet.

yep, fine 👍

Reply:
> To my mind it's about providing descriptions to data

Agree. There's already the `meta` field for general user info about the .dvc file itself.

> would -m be a better option? to remind git commit -m

Good idea but I also think it's not intuitive: It could be confusing since this is `add`, not `commit`.

> Both -d and -m are taken.

Not for `add`.

Reply:
My suggestion:
```suggestion
- `--desc <text>` - arbitrary description for the data `targets`.
```
> Please confirm metavar per https://github.com/iterative/dvc/pull/4901#pullrequestreview-532809188)

BTW does the same `desc` get applied to multiple targets (if several are provided to `dvc add`)? Thanks

Reply:
Thanks, updated the doc and the help in the original PR.

> BTW does the same desc get applied to multiple targets (if several are provided to dvc add)? Thanks

Yes.